File: 문자열 압축.py
def solution(s):
    answer = len(s)
    for i in range(1, len(s) // 2 + 1):
        temp = s[0:i]
        length = len(s)
        same = 1
        for j in range(i, len(s), i):
            if s[j:j + i] == temp:
                length -= i
                same += 1
            else:
                if same > 1:
                    length += len(str(same))
                temp = s[j:j + i]
                same = 1
        if same > 1:
            length += len(str(same))
        answer = min(length, answer) 
    return answer

File: kotlin/문자열압축.kt
class Solution {
    fun solution(s: String): Int {
        var answer = s.length;
        for (i in 1 until (s.length / 2 + 1)) {
          var temp = s.substring(0,i);
          var length = s.length;
          var same = 1;
          for (j in i until (s.length) step(i)){
            if (s.substring(j, Math.min(j + i, s.length)) == temp) {
              length = length - i;
              same += 1;
            } else {
              if (same > 1) {
                length += same.toString().length;
              }
              temp = s.substring(j, Math.min(j + i, s.length));
              same = 1;
            }
          }
          if (same > 1) {
            length += same.toString().length;
          }
          answer = Math.min(length, answer);
        }
        return answer
    }
}